### PR TITLE
echidna: init at 1.7.2

### PR DESCRIPTION
**Pull request by arcz**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by samb96** [comment]
@GrahamcOfBorg eval

File: pkgs/tools/security/echidna/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    substituteInPlace package.yaml \
+      --replace 'condition: os(linux)' 'condition: false'
+  '';
+  preConfigure = "hpack";
+  # tests require specific version of solc
+  doCheck = false;
+
+  description = "A Fast Smart Contract Fuzzer";
+  homepage = "https://github.com/crytic/echidna";
+  license = lib.licenses.agpl3;
```
Comment (rmcgibbo) [inline pkgs/tools/security/echidna/default.nix:42]: agpl3Plus or agpl3Only?

**Review by rmcgibbo — CHANGES_REQUESTED** [review]
Result of `nixpkgs-review pr 106919` at 7ecf5656 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build (new failures):</summary>
  <ul>
    <li>echidna</li>
  </ul>
</details>

build failure: https://gist.github.com/rmcgibbo/8c138763bfa8b9555c93d953bebd1a14

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -1161,6 +1161,9 @@ in
 
   sedutil = callPackage ../tools/security/sedutil { };
 
+  echidna = haskell.lib.justStaticExecutables
+    (haskellPackages.callPackage ../tools/security/echidna { });
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:1165]:
```suggestion
  echidna = haskell.lib.justStaticExecutables (haskellPackages.callPackage ../tools/security/echidna { });
```

**Comment by arcz** [comment]
@rmcgibbo @SuperSandro2000 I updated this PR, can you please take a look?

**Comment by rmcgibbo** [comment]
I don't see any problems, but I'm not familiar with the nixpkgs haskell ecosystem at all.

File: pkgs/tools/security/echidna/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  shellHook = "hpack";
+  license = lib.licenses.agpl3Plus;
+  doHaddock = false;
+  # tests depend on a specific version of solc
+  doCheck = false;
+
+  description = "Ethereum smart contract fuzzer";
+  homepage = "https://github.com/crytic/echidna";
+  maintainers = with lib.maintainers; [ arturcygan ];
+  platforms = lib.platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/tools/security/echidna/default.nix:53]:
```suggestion
  doHaddock = false;
  # tests depend on a specific version of solc
  doCheck = false;

  description = "Ethereum smart contract fuzzer";
  homepage = "https://github.com/crytic/echidna";
  license = lib.licenses.agpl3Plus;
  maintainers = with lib.maintainers; [ arturcygan ];
  platforms = lib.platforms.unix;
```

File: pkgs/tools/security/echidna/default.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+, wl-pprint-annotated, word8, yaml , extra, ListLike, semver
+}:
+mkDerivation rec {
+  pname = "echidna";
+  version = "1.7.2";
+
+  src = fetchFromGitHub {
+    owner = "crytic";
+    repo = "echidna";
+    # this commit is on top of 1.7.2, updates hevm
```
Comment (SuperSandro2000) [inline pkgs/tools/security/echidna/default.nix:19]: We don't want to do this. People think this is the 1.7.2 release.

**Comment by vbgl** [comment]
FWIW, the `echidna` package was broken by 06274cd85dede525fec5c992d0133a8758f96f8f (part of #134317).

**Pull request merged**